control-plane-operator/controllers/hostedcontrolplane/v2/configoperator/podmonitor.go

```diff
@@ -0,0 +1,17 @@
+package configoperator
+
+import (
+	component "github.com/openshift/hypershift/support/controlplane-component"
+	"github.com/openshift/hypershift/support/metrics"
+	"github.com/openshift/hypershift/support/util"
+
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+)
+
+func adaptPodMonitor(cpContext component.ControlPlaneContext, podMonitor *prometheusoperatorv1.PodMonitor) error {
+	podMonitor.Spec.NamespaceSelector = prometheusoperatorv1.NamespaceSelector{MatchNames: []string{cpContext.HCP.Namespace}}
```

Comment:
If we're going to do similar adapts for pod/service monitors, maybe it makes sense to have a common function to call

Reply:
done